### DISCUSSION TITLE
Muting / adjusting volume on individual channels

### DISCUSSION THREAD

**Discussion by stealthrabbi** [katspaugh/wavesurfer.js#2740] [Q&A] [answered]
I'm using wavsurfer to play mono and stereo files. if I have a stereo file, but wish to only play 1 of the two channels, is there a clean way to do this? Any guidance is appreciated.

Does this involve doing something with gain? Code is currently applying gain as a filter, to allow playing audio at louder than 100%.

**Comment by katspaugh** [comment]
You can pan all the way to that channel. See this example: https://wavesurfer-js.org/example/panner/index.html

**Reply by stealthrabbi** [reply]
Thanks. So, just to be clear, if an audio file is dual channel, and I use the panner to select all the way to the left, I will only hear the left channel (as opposed to just pushing both channels of sound to the left speaker). Hopefully that question makes sense.

**Reply by katspaugh** [reply]
Good question! I’ve played too much modular synths and automatically assumed the former, but it might be not the case. Reading up.

**Reply by katspaugh** [reply]
Funnily enough, googling for “web audio mute one channel” gives this as the first result, which seems to be the perfect answer: https://github.com/wavesurfer-js/wavesurfer.js/issues/1497#issuecomment-435636656
Tl;dr use a channel splitter

**Reply by stealthrabbi** [reply]
So there is also a StereoPanner. The example you linked to initally (the panner example) shows setting a property on the wavesurfer object called `panner`, but I don't see such one. Maybe the example is based on an older version of the library.

**Reply by katspaugh** [reply]
It's not a wavesurfer object, it just uses wavesurfer's AudioContext (wavesurfer.backend.ac) to create a regular Web Audio panner node. You can create your own AudioContext, too.

**Comment by prakharpbuf** [comment, marked as answer]
There a [channel splitter node](https://developer.mozilla.org/en-US/docs/Web/API/ChannelSplitterNode) in the web audio api.
This node separates the different channels of an audio source into a set of mono outputs. This is useful for accessing each channel separately.

What you will want to do is 
1.create a channel splitter node with proper [parameters](https://developer.mozilla.org/en-US/docs/Web/API/ChannelSplitterNode/ChannelSplitterNode#parameters) 
2. create a [gain node](https://developer.mozilla.org/en-US/docs/Web/API/GainNode) which [connects](https://developer.mozilla.org/en-US/docs/Web/API/AudioNode/connect) to the required outputIndex of your splitter node
3. wavesurfer.setFilters([]) to get the output

It might be a little bit more complicated than that. Please try it out and let me know if you get stuck, I'll be happy to help!

**Reply by stealthrabbi** [reply]
So the idea is that I maintain 2 separate gain nodes, each one connectined to one of hte output indicies from teh splitter node, and then I adjust the gain on the corresponding gain node if I want to play just left or right channel?

Or, do I keep just a single gain node, but connect it to either one or both of the output index? Do I need to create a separte gain node whenever the user chooses to hear a left vs right vs both channels? I see connect() says it only accepts a single channel index. I'm also not sure how to use this paradigm and have it play all channels. Perhaps I'd have to recreate the gain node w\o a splitter node?

**Reply by prakharpbuf** [reply]
> So the idea is that I maintain 2 separate gain nodes, each one connectined to one of hte output indicies from teh splitter node, and then I adjust the gain on the corresponding gain node if I want to play just left or right channel?

Right, this is one way to go. This will allow you to listen and control the gain of each channel separately.

The other way is to only connect a gain node to the required outputIndex of the splitter node and ignore all other output indices. Go this route if you only care about a single channel.

**Reply by stealthrabbi** [reply]
I ended up going with the 2 gains. This enabled us to do reduced volume playback on a particular channel and works great. Thanks for the assistance!

**Comment by jenga555** [comment]


**Reply by katspaugh** [reply]
You can still use WebAudio like this: https://wavesurfer-js.org/examples/?webaudio.js
